_posts/2023-05-17-otp-26-highlights.md

```diff
@@ -0,0 +1,892 @@
+list_to_binary(           list_to_bitstring(        list_to_existing_atom(
+list_to_float(            list_to_integer(          list_to_pid(
+list_to_port(             list_to_ref(              list_to_tuple(
+Press tab to see all 37 expansions
+```
+
+Pressing TAB again, more BIFs are shown, as well as possible shell commands
+and modules:
+
+```
+10> l
+bifs
+length(                   link(                     list_to_atom(
+list_to_binary(           list_to_bitstring(        list_to_existing_atom(
+list_to_float(            list_to_integer(          list_to_pid(
+list_to_port(             list_to_ref(              list_to_tuple(
+load_module(
+commands
+l(     lc(    lm(    ls(
+modules
+lcnt:                      leex:                      lists:
+local_tcp:                 local_udp:                 log_mf_h:
+logger:                    logger_backend:            logger_config:
+logger_disk_log_h:         logger_filters:            logger_formatter:
+logger_h_common:           logger_handler_watcher:    logger_olp:
+logger_proxy:              logger_server:             logger_simple_h:
+logger_std_h:              logger_sup:
+```
+
+Typing `ists:` and pressing TAB, a partial list of functions in the
+`lists` modules are shown:
+
+```
+10> lists:
+functions
+all(            any(            append(         concat(         delete(
+droplast(       dropwhile(      duplicate(      enumerate(      filter(
+filtermap(      flatlength(     flatmap(        flatten(        foldl(
+foldr(          foreach(        join(           keydelete(      keyfind(
+Press tab to see all 72 expansions
+```
+
+Typing `m` and pressing TAB, the list of functions is narrowed down to
+just those beginning with the letter `m`:
+
+```
+10> lists:m
+functions
+map(            mapfoldl(       mapfoldr(       max(            member(
+merge(          merge3(         min(            module_info(
+```
+
+# Improvements of maps
+
+## Changed ordering of atom keys
+
+In OTP 25 and earlier releases, when printing small maps (up to 32 elements)
+having atom keys, the elements are printed sorted according to the term
+order of their keys:
+
+```erlang
+1> AM = #{a => 1, b => 2, c => 3}.
+#{a => 1,b => 2,c => 3}
+2> maps:to_list(AM).
+[{a,1},{b,2},{c,3}]
+```
+
+In OTP 26, as an optimization for certain map operations, such as
+`maps:from_list/1`, maps with atom keys are now sorted in a different
+order. The new order is unpredictable and may change between different
+invocations of the Erlang VM. On my computer at the time of writing,
+I got the following order:
+
+```erlang
+1> AM = #{a => 1, b => 2, c => 3}.
+#{c => 3,a => 1,b => 2}
+2> maps:to_list(AM).
+[{c,3},{a,1},{b,2}]
+```
+
+There is a new modifier `k` for format strings to specify that maps should
+be sorted before printing:
+
+```erlang
+3> io:format("~kp\n", [AM]).
+#{a => 1,b => 2,c => 3}
+ok
+```
+
+It is also possible to use a custom ordering fun. For example, to
```

Comment:
"custom ordering fun": from the example I can guess how it works, but I don't think a link to the documentation would hurt.

Reply:
I can only link to the beginning of `io:format()`; I've done that.